app/bundles/CampaignBundle/Assets/js/campaign.js

```diff
@@ -2330,3 +2333,45 @@ Mautic.previewCampaignLabels = function() {
         });
     });
 }
+
+Mautic.campaignAuditlogOnLoad = function (container, response) {
+    mQuery("#campaign-auditlog a[data-activate-details='all']").on('click', function() {
+        let $icon = mQuery(this).find('span').first();
+        if ($icon.hasClass('ri-arrow-down-s-line')) {
+            mQuery("#campaign-auditlog a[data-activate-details!='all']").each(function () {
+                const detailsId = mQuery(this).data('activate-details');
+                if (detailsId && mQuery('#auditlog-details-' + detailsId).length) {
+                    mQuery('#auditlog-details-' + detailsId).removeClass('hide');
+                    mQuery(this).addClass('active');
+                }
+            });
+            $icon.removeClass('ri-arrow-down-s-line').addClass('ri-arrow-up-s-line');
+        } else {
+            mQuery("#campaign-auditlog a[data-activate-details!='all']").each(function () {
+                const detailsId = mQuery(this).data('activate-details');
+                if (detailsId && mQuery('#auditlog-details-' + detailsId).length) {
+                    mQuery('#auditlog-details-' + detailsId).addClass('hide');
+                    mQuery(this).removeClass('active');
+                }
+            });
+            $icon.removeClass('ri-arrow-up-s-line').addClass('ri-arrow-down-s-line');
+        }
+    });
```

Comment:
This is just a small optimization but just so you'd know about it. Every time you create a jQuery object it does quite an expensive operation of finding the right HTML element(s). So it's better to find it once and store it in a variable than finding it over and over.

Notice that for example `mQuery('#auditlog-details-' + detailsId)` could be stored into a variable and used in other places.

Reply:
Thanks @escopecz for pointing that out. That's a good point. 

For simple stuff like selecting an element, I would even go one step further and don't use mQuery. So that we can get rid of this dependency at some point. But that might just be my point of view.

So it would be something like:
const auditlog = document.getElementById('#auditlog-details-' + detailsId);

Reply:
Excelent point!